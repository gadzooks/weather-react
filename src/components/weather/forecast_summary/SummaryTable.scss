// Place all the styles related to the Weather controller here.
// They will automatically be included in application.css.
// You can use Sass (SCSS) here: http://sass-lang.com/
// .weather-forecast-table {
//   font-family: 'Courier';
//   // font-family: 'Apercu Mono'
//   border-collapse: collapse;
//   margin: 25px 0;
//   font-size: 0.9em;
//   font-family: sans-serif;
//   min-width: 400px;
//   box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
// }

// $color-alert-bg: black;

// .alert-cell-triangle {
//     position: relative;
// }
// .alert-cell-triangle:after {
//   content: "";
//   position: absolute;
//   top: 0;
//   right: 0;
//   width: 0;
//   height: 0; 
//   display: block;
//   border-left: 15px solid transparent;
//   border-bottom: 15px solid transparent;
//   border-top: 15px solid $color-alert-bg;
// }

// .alert-details {
//   padding: 0.5rem;
//   margin-bottom: 1.25rem;
//   border: 2px solid #eee;
//   border-left-width: .25rem;
//   border-radius: .5rem;
//   span.title {
//     margin-top: 0;
//     margin-bottom: .25rem;
//     font-weight: bolder;
//     padding: 5px;
//     border-radius: .25rem;
//   }
//   .regions {
//     padding: 3px;
//     span.region {
//       font-weight: 700;
//     }
//   }
//   .description {
//     padding: 3px;
//     span.description {
//       font-weight: 700;
//     }
//   }
// }


// // level 1 warning
// $color-alert-advisory: rgba(23, 188, 255, 1);
// .alert-advisory {
//   border-left-color: $color-alert-advisory;
// }

// // level 2 warning
// $color-alert-watch: rgba(255, 213, 46, 1);
// .alert-watch {
//   border-left-color: $color-alert-watch;
// }

// $color-alert-warning: rgba(255, 66, 39, 0.88);
// // level 3 warning
// .alert-warning {
//   border-left-color: $color-alert-warning;
// }

// span.alerts-summary {
//   background-color: blue;
//   color: white;
//   padding: 1px 5px;
//   margin-right: 10px;
//   border-radius: .2rem;
// }

// span.alert_id {
//   background-color: $color-alert-bg;
//   padding: 1px 5px;
//   margin: 3px;
//   a {
//     color: white;
//   }
//   border-radius: .2rem;
// }

// table.weather-forecast-alerts {
//   tr.table-heading {
//     td {
//       border: 0;
//       font-size: 1.2em;
//     }
//   }
// }

table.table {
  border-collapse: collapse;
  font-size: 0.9em;
  font-family: sans-serif;
  min-width: 400px;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
}

.styled-table thead tr {
  background-color: #f9d6a7;
  color: black;
  font-weight: 5em;
  // text-align: center;
}

.styled-table th,
.styled-table td {
  padding: 12px 8px;
  text-align: center;
}

.styled-table td.location-name {
  text-align: left;
}

.forecast-button {
  background: none !important;
  border: none;
  padding: 0 !important;
  /*optional*/
  font-family: arial, sans-serif;
  /*input has OS specific font-family*/
  color: #069;
  text-decoration: underline;
  cursor: pointer;
}

// < !-- HTML !--><button class="button-2" role="button">Button 2</button>
// https: //getcssscan.com/css-buttons-examples
/* CSS */
.button-2 {
  background-color: rgb(51 51 51 / 10%);
  border-radius: 8px;
  border-width: 0;
  color: #333333;
  cursor: pointer;
  display: inline-block;
  font-family: "Haas Grot Text R Web", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 500;
  line-height: 20px;
  list-style: none;
  margin: 0;
  padding: 10px 12px;
  text-align: center;
  transition: all 200ms;
  vertical-align: baseline;
  white-space: nowrap;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
}

// FIXME : not being used currently
table.weather-forecast-summary-11111 {
  tbody {
    white-space: nowrap;
    td {
      padding: 0.45em;
    }
  }
  td.weather-cell {
    border: 2px solid rgba(1,2,4,0.07);
  }
  td.center {
    text-align: center;
  }
  thead.table-heading {
    td {
      background-color: #f08e07;
      color: #ffffff;
      text-align: center;
    }
  }

  tr.planetory-info {
    span.sunrise {
      i {
        color: #ffc107;
        padding-right: 0.4em;
      }
      padding-right: 2em;
    }
    span.sunset {
      i {
        color: #0e151d;
        padding-right: 0.4em;
      }
      padding-left: 2em;
    }
  }
}

.wta-link {
  padding-right: 5px;
}

.all-alerts {
  padding-top: 10px;
}

.alert-for-this-day {
  border-bottom: 1px solid red;
}

// table.weather-forecast {
//   border-collapse:collapse;
//   tr.weather-cell td {
//     border: 2px solid rgba(1, 2, 4, 0.07);
//   }

//   thead {
//     tr.header-secondary-main {
//       background-color:#f2d8a4;
//     }
//   }

//   tbody {
//     white-space: nowrap;

//     tr.daily-summary {
//       background-color: #feffd4;
//     }

//     .weekend {
//       background-image:linear-gradient(142deg, #b2f7c6 11.11%, #faf5f5 11.11%, #faf5f5 50%, #b2f7c6 50%, #b2f7c6 61.11%, #faf5f5 61.11%, #faf5f5 100%);
//       background-size: 14.62px 11.42px;
//     }

//     tr.region-details {
//       td {
//         column-span: all;
//         text-align: center;
//       }
//       background: #fef7e9;
//     }

//     tr.weather-details {
//       span.forecast-id {
//         color: white;
//         background-color: rgba(156, 191, 240, 0.8);
//         padding: 3px 3px 3px 5px;
//         border-radius: 0.25em;
//         margin-right: 5px;
//       }

//       td.summary {
//         text-align: left;
//       }

//       td.location-row {
//         text-align: left;
//       }

//       td.left-col-of-merged-cells {
//         border-right: solid 1px #FFF;
//       }

//       td.right-col-of-merged-cells {
//         border-left: solid 1px #FFF;
//       }

//       td.forecast-date span.day-of-month {
//         border:none;
//         color: #df788e;
//         font-size: 0.85em;
//         font-weight: 900;
//         white-space: nowrap;
//       }

//       td.precipitation {
//         text-align: center;
//       }

//       td.precipitation-prob {
//         border-right: none;
//         text-align: right;
//       }
//       td.precipitation-amount {
//         border-left: none;
//         text-align: right;
//       }
//       td.cloud-cover {
//         text-align: right;
//         padding-right: 4em;
//       }
//     }
//   }

//   .header-main {
//     background-color:#def0ff;
//   }

//   tr.header-secondary {
//     background-color:#fef7e9;
//     text-align: center;
//     td {
//       padding: 8px 8px;
//     }
//   }
// }

.align-right {
  text-align: right;
}

.weather-icon {
  font-size: 1.4em;
  font-weight: 1.2em;
}

a.freepik {
  text-decoration: underline;
  font-size: 0.8em;
}

@media (max-width: 500px) {
  .styled-table th,
  .styled-table td {
    padding: 6px 0px;
    text-align: center;
  }

  .weather-icon {
    font-size: 1.1em;
  }

  table.table {
    font-size: 0.8em;
  }
 
  .forecast-button {
    font-size: 0.8em;
  }

  .styled-table td.location-name {
    padding-left: 12px;
  }

}