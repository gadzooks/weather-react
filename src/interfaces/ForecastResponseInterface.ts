import { LocationInterface } from './LocationInterface';
import { RegionInterface } from './RegionInterface';
import { DailyForecastInterface } from './DailyForecastInterface';

export interface RegionById {
    [key: string]: RegionInterface;
}

export interface RegionsById {
  byId: RegionById;
  allIds: string[];
}

export interface LocationById {
  [key: string]: LocationInterface;
}

export interface ForecastById {
  [key: string]: DailyForecastInterface[];
}

export interface ForecastsById {
  byId: ForecastById;
}

export interface LocationsById {
  byId: LocationById;
  allIds: string[];
}

export interface ForecastResponse {
  dates: string[];
  regions: RegionsById;
  locations: LocationsById;
  forecasts: ForecastsById;
}
export interface ForeacastDates {
  dates: string[];
  parsedDates: Date[];
  weekends: boolean[];
}

export const DefaultForecastDates: ForeacastDates = {
  dates: [],
  parsedDates: [],
  weekends: [],
};

export interface ForecastResponseStatus {
  isLoaded?: boolean;
  error: Error | null;
  forecast: ForecastResponse | null;
  forecastDates: ForeacastDates;
}

export const DefaultForecastResponseStatus: ForecastResponseStatus = {
  isLoaded: false,
  error: null,
  forecast: null,
  forecastDates: DefaultForecastDates,
};

// dates: [
//     "2021-04-17T07:00:00.000+00:00",
//     "2021-04-18T07:00:00.000+00:00",
//     "2021-04-19T07:00:00.000+00:00",
//     "2021-04-20T07:00:00.000+00:00",
// ],
// regions: {
//     byId: {
//     central_cascades: {
//         id: "central_cascades",
//         description: "Central Cascades",
//         regionId: "central_cascades",
//         locations: ["renton", "yakima"],
//     },
//     western_wa: {
//         id: "western_wa",
//         description: "Western WA",
//         regionId: "western_wa",
//         locations: ["seattle", "bellevue"],
//     },
//     },
//     allIds: ["central_cascades", "western_wa"],
// },
// locations: {
//     byId: {
//     renton: {
//         id: "renton",
//         description: "Renton",
//         regionId: "central_cascades",
//     },
//     yakima: {
//         id: "yakima",
//         description: "Yakima",
//         regionId: "central_cascades",
//     },
// forecasts: {
//     byId: {
//       renton: [
//         {
//           time: "2021-04-17T07:00:00.000+00:00",
//           summary: "Clear conditions throughout the day.",
//           icon: "day-hail",
//           precipProbability: 0.0,
//           temperature: 60.9,
//           apparentTemperature: 60.9,
//           dewPoint: 40.1,
//           visibility: 120,
//           cloudCover: 1,
//           temperatureHigh: 74.9,
//           temperatureLow: 47.3,
//           sunsetTime: 1618714951,
//           sunriseTime: 1618665359,
//           precipAmount: 0.0,
//         },
