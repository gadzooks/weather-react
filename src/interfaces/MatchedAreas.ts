import { LocationInterface } from './LocationInterface';
import { RegionInterface } from './RegionInterface';

export interface MatchedAreas {
  regions?: RegionInterface[];
  locationsByRegion?: {
    [regionName: string]: LocationInterface[];
  };
  // dates?: string[];
  totalMatchedRegions: number;
}
