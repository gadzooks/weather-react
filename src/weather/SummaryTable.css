.SummaryTable {
}

table tr th {
    background-color: #f2e2a4d6;
}