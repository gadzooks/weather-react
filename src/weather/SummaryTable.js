import Region from './Region.js';
import './SummaryTable.css';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableCell from '@mui/material/TableCell';
import {formatDate} from '../utils/date'

function SummaryTable(props) {
  const inputs = props.inputs || {
    regionalForecast: [
      {
        name: "central_cascades",
        description: "Central Cascades",

        locations: [
          {
            name: "renton",
            description: "Renton",
            region: "central_cascades",
            daily: {
              type: "daily",
              summary:
                "Cooling down with a chance of rain Thursday, Friday \u0026 Saturday.",
              icon: null,
              data: [
                {
                  time: "2021-04-17T07:00:00.000+00:00",
                  summary: "Clear conditions throughout the day.",
                  icon: "day-hail",
                  precipProbability: 0.0,
                  temperature: 60.9,
                  apparentTemperature: 60.9,
                  dewPoint: 40.1,
                  visibility: 120,
                  cloudCover: 1,
                  temperatureHigh: 74.9,
                  temperatureLow: 47.3,
                  sunsetTime: 1618714951,
                  sunriseTime: 1618665359,
                  precipAmount: 0.0,
                },
                {
                  time: "2021-04-18T07:00:00.000+00:00",
                  summary: "Clear conditions throughout the day.",
                  icon: "day-haze",
                  precipProbability: 0.0,
                  temperature: 62.1,
                  apparentTemperature: 62.1,
                  dewPoint: 41.4,
                  visibility: 150,
                  cloudCover: 14,
                  temperatureHigh: 72.9,
                  temperatureLow: 52.1,
                  sunsetTime: 1618801436,
                  sunriseTime: 1618751647,
                  precipAmount: 0.0,
                },
                {
                  time: "2021-04-19T07:00:00.000+00:00",
                  summary: "Clear conditions throughout the day.",
                  icon: "day-storm-showers",
                  precipProbability: 0.0,
                  temperature: 60.4,
                  apparentTemperature: 60.4,
                  dewPoint: 39.9,
                  visibility: 150,
                  cloudCover: 18,
                  temperatureHigh: 70.1,
                  temperatureLow: 52.1,
                  sunsetTime: 1618887921,
                  sunriseTime: 1618837937,
                  precipAmount: 0.0,
                },
                {
                  time: "2021-04-20T07:00:00.000+00:00",
                  summary: "Partly cloudy throughout the day.",
                  icon: "hot",
                  precipProbability: 19.0,
                  temperature: 58.1,
                  apparentTemperature: 58.1,
                  dewPoint: 44.4,
                  visibility: 140,
                  cloudCover: 33,
                  temperatureHigh: 65.9,
                  temperatureLow: 51.0,
                  sunsetTime: 1618974405,
                  sunriseTime: 1618924228,
                  precipAmount: 0.0,
                },
              ],
            },
          },
          {
            name: "yakima",
            description: "Yakima",
            region: "central_cascades",
          },
        ],
      },
      {
        name: "central_wa",
        description: "Central Washington",
        locations: [
          {
            name: "seattle",
            description: "Seattle City",
            region: "central_wa",
          },
          {
            name: "bellevue",
            description: "Bellevue City",
            region: "central_wa",
          },
        ],
      },
    ],

    dates: [
      "2021-04-17T07:00:00.000+00:00",
      "2021-04-18T07:00:00.000+00:00",
      "2021-04-19T07:00:00.000+00:00",
      "2021-04-20T07:00:00.000+00:00",
    ],
  };

  const dates = inputs.dates || ["SAT JUN 04", "SUN JUN 05", "MON JUN 06"];
  const regions = inputs.regionalForecast;
  return (
    <>
      <TableHead>
        <TableRow>
          <TableCell>Weather Alerts</TableCell>
          <TableCell>Location</TableCell>
          {dates.map((date) => {
            return <TableCell>{formatDate(date)}</TableCell>;
          })}
        </TableRow>
      </TableHead>
      {regions.map((r) => {
        return <Region region={r} key={r.name} />;
      })}
    </>
  );
}

export default SummaryTable;