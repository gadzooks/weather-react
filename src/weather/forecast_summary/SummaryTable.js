import Region from './Region.js';
import './SummaryTable.scss';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableCell from '@mui/material/TableCell';
import { Table } from '@mui/material';
import { parse, format } from 'fecha';

function matchedOne(needle, haystack) {
  const names = haystack.map((l) => {return l.name});
  return names.find((element) => element.match(needle));
}

function SummaryTable(props) {
  const inputs = props.inputs;
  const isWeekend = props.isWeekend;
  const dates = props.dates;
  const regionIds = inputs.regions.allIds;
  return (
    <>
      <Table className='table table-sm weather-forecast-summary'>
        <TableHead className='table-heading'>
          <TableRow>
            <TableCell>Weather Alerts</TableCell>
            <TableCell>Location</TableCell>
            {dates.map((date) => {
              return <TableCell key={date}>{format(date, 'ddd MMM DD').toUpperCase()}</TableCell>;
            })}
          </TableRow>
        </TableHead>
        {regionIds.map((id) => {
          const region = inputs["regions"]["byId"][id];
          if (matchedOne(props.searchText, region.locations)) {
            return (
              <Region
                id={id}
                inputs={inputs}
                key={id}
                searchText={props.searchText}
                isWeekend={isWeekend}
              />
            );
          } else {
            return null;
          }
        })}
      </Table>
    </>
  );
}

export default SummaryTable;