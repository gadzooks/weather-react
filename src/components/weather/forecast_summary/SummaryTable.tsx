import './SummaryTable.scss';
import { format } from 'fecha';
import React from 'react';
import { LocationInterface } from '../../../interfaces/LocationInterface';
import { ForeacastDates, ForecastResponse } from '../../../interfaces/ForecastResponseInterface';
import Region from './Region';
import { RegionInterface } from '../../../interfaces/RegionInterface';
import { MatchedAreas } from '../../../interfaces/MatchedAreas';
import { DailyForecastFilter, dateSelectedMatchesForecastDates, matchesSelecteDate } from '../../../interfaces/DailyForecastFilter';
import alertsFound from '../../../utils/count';

// eslint-disable-next-line max-len
export function matchedLocations(needle: RegExp | null, region: RegionInterface) :LocationInterface[] {
  if (!needle) return region.locations;
  return region.locations.filter((l) => l.description.match(needle));
}

export interface SummaryTableProps {
  forecastResponse: ForecastResponse | null;
  forecastDates: ForeacastDates;
  matchedAreas: MatchedAreas;
  dailyForecastFilter: DailyForecastFilter;
  setDailyForecastFilter: any;
  setForecastDetailsForLocation: any;
}

function prevDateWithinRange(
  date: Date | null,
  index: number,
  dates: (Date | null)[],
): string | null {
  if (date === null) return null;
  if (index === 0) return null;
  const prevParsedDate = dates[index - 1];
  if (!prevParsedDate) return null;
  return format(prevParsedDate, 'YYYY-MM-DD');
}

function nextDateWithinRange(
  date: Date | null,
  index: number,
  dates: (Date | null)[],
): string | null {
  if (date === null) return null;
  if (index === (dates.length - 1)) return null;
  const nextParsedDate = dates[index + 1];
  if (!nextParsedDate) return null;
  return format(nextParsedDate, 'YYYY-MM-DD');
}

function SummaryTable(props: SummaryTableProps) {
  const { forecastDates } = props;
  const { weekends } = forecastDates;
  const { parsedDates } = forecastDates;
  const { matchedAreas } = props;
  const regions = matchedAreas.regions || [];
  const locationsByRegion = matchedAreas.locationsByRegion || {};
  const { dailyForecastFilter } = props;
  const { forecastResponse } = props;
  const { setDailyForecastFilter } = props;
  const { setForecastDetailsForLocation } = props;
  const alertsById = forecastResponse?.alertsById || null;

  const dateSelectedIsWithinForecastRange = dateSelectedMatchesForecastDates(
    forecastDates.dates,
    dailyForecastFilter.date,
  );

  const selectDate = (date: string) => {
    const dFF = { ...dailyForecastFilter } as DailyForecastFilter;
    if (date === dFF.date) {
      dFF.date = '';
    } else {
      dFF.date = date;
    }
    setDailyForecastFilter(dFF);
  };

  // console.log(JSON.stringify(alertsById));

  return (
    <table className='table styled-table'>
      <thead className='table-heading'>
        {/* <tr>
          <td rowSpan={2}>Location</td>
          {parsedDates.map((date) => {
            const txt = date === null ? '' : format(date, 'ddd').toUpperCase();
            const dateMatches = matchesSelecteDate(
              date,
              dailyForecastFilter.date,
            );

            return (
              (!dateSelectedIsWithinForecastRange || dateMatches) && (
              <td key={txt} align='center'>
                {txt}
              </td>
              )
            );
          })}
        </tr> */}

        <tr>
          {alertsFound(alertsById) && <td>Alerts</td> }
          <td>Location</td>
          {parsedDates.map((date, index) => {
            const txt = date === null ? '' : format(date, 'DD').toUpperCase();
            const dateKey = date === null ? '' : format(date, 'YYYY-MM-DD').toUpperCase();
            const dateMatches = matchesSelecteDate(
              date,
              dailyForecastFilter.date,
            );
            const prevDateKey = prevDateWithinRange(date, index, parsedDates);
            const nextDateKey = nextDateWithinRange(date, index, parsedDates);
            return (
              (!dateSelectedIsWithinForecastRange || dateMatches) && (
              <td key={txt} align='center'>
                {dateSelectedIsWithinForecastRange && (
                <>
                  <button
                    className='button-2 left-arrow'
                    type='button'
                    onClick={() => selectDate(prevDateKey || '')}
                  >
                    &larr;
                  </button>
                  <button
                    type='button'
                    className='button-2 forecast-date'
                    onClick={() => selectDate(dateKey)}
                  >
                    {txt}
                  </button>
                  <button
                    className='button-2 right-arrow'
                    type='button'
                    onClick={() => selectDate(nextDateKey || '')}
                  >
                    &rarr;
                  </button>
                </>
                )}
                {!dateSelectedIsWithinForecastRange && (
                <button
                  type='button'
                  className='button-2'
                  onClick={() => selectDate(dateKey)}
                >
                  {txt}
                </button>
                )}
              </td>
              )
            );
          })}
        </tr>
      </thead>
      {regions.map((region: RegionInterface) => {
        const locations = locationsByRegion[region.name];
        if (forecastResponse?.forecasts) {
          return (
            <Region
              key={region.name}
              isWeekend={weekends}
              region={region}
              forecastsById={forecastResponse?.forecasts}
              locations={locations}
              dailyForecastFilter={dailyForecastFilter}
              dateSelectedIsWithinForecastRange={
                  dateSelectedIsWithinForecastRange
                }
              setForecastDetailsForLocation={setForecastDetailsForLocation}
              alertsById={alertsById}
            />
          );
        }
        return null;
      })}
    </table>
  );
}

export default SummaryTable;
