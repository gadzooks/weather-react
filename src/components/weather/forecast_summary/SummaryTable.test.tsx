import React from 'react';
import renderer from 'react-test-renderer';
import mockWeatherForecastNormalized from '../../../api/mockData';
import { ForecastResponse } from '../../../interfaces/ForecastResponseInterface';
import { MatchedAreas } from '../../../interfaces/MatchedAreas';
import findMatchedAreas from '../../../utils/filterMatchedAreas';
import SummaryTable, { SummaryTableProps } from './SummaryTable';

it('renders a snapshot', () => {
  const forecast = mockWeatherForecastNormalized()
    .data as unknown as ForecastResponse;

  const matchedAreas:MatchedAreas = findMatchedAreas(null, forecast.regions);
  const weatherPageArgs: SummaryTableProps = {
    dailyForecastFilter: {},
    matchedAreas,
    forecastResponse: forecast,
    setDailyForecastFilter: () => {},
    setForecastDetailsForLocation: () => {},
  };
  const tree = renderer
    .create(<SummaryTable {...weatherPageArgs} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
