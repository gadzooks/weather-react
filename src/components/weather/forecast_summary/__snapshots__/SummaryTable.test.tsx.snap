// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<table
  className="table table-sm weather-forecast-summary"
>
  <thead
    className="table-heading"
  >
    <tr>
      <td>
        Location
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SAT 17
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SUN 18
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          MON 19
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          TUE 20
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          WED 21
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          THU 22
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          FRI 23
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SAT 24
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SUN 25
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          MON 26
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          TUE 27
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          WED 28
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          THU 29
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          FRI 30
        </button>
      </td>
      <td
        align="center"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SAT 01
        </button>
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      className="region-details"
    >
      <td
        align="center"
        colSpan={16}
      >
        Cities
      </td>
    </tr>
    <tr
      className="weather-cell"
    >
      <td
        className="weather-cell"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          SEATTLE CITY
        </button>
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-day-cloudy day-cloudy "
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-day-cloudy day-cloudy "
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with morning rain."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with a chance of rain throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with rain in the morning and afternoon."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Cloudy skies throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Cloudy skies throughout the day."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-10"
          title="Partly cloudy throughout the day."
        />
      </td>
    </tr>
    <tr
      className="weather-cell"
    >
      <td
        className="weather-cell"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          BONNEY LAKE
        </button>
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-day-sunny day-sunny "
          title="Clear conditions throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with rain in the morning and afternoon."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-day-cloudy day-cloudy "
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with rain."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with rain."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Partly cloudy throughout the day with a chance of rain."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-rain rain "
          title="Cloudy skies throughout the day with rain in the morning and afternoon."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Partly cloudy throughout the day."
        />
      </td>
      <td
        className="weather-cell center  "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-50"
          title="Cloudy skies throughout the day."
        />
      </td>
      <td
        className="weather-cell center  weekend "
      >
        <i
          className="wi weather-icon wi-cloudy cloudy sunshine-10"
          title="Clearing in the afternoon."
        />
      </td>
    </tr>
  </tbody>
</table>
`;
